### PR TITLE
feat: [SRM-14035]: addressing templates scenarios for ELK

### PR DESCRIPTION
**Pull request by harshilbits**
### Summary
1. This PR is to ensure all templates scenarios works for ELK.
2. Whatever new components we have made for ELK should have template support .
3. Refactoring to ensure same functions are getting used in verify step , run pipeline form and monitored service creation from template (to show the labels and validation error messages).

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/customiseHealthSource/CustomiseHealthSource.utils.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export const LoadSourceByType = ({
 }): JSX.Element | null => {
   const isSplunkMetricEnabled = useFeatureFlag(FeatureFlag.CVNG_SPLUNK_METRICS)
   const isSumoLogicEnabled = useFeatureFlag(FeatureFlag.SRM_SUMO)
-  const isELKEnabled = useFeatureFlag(FeatureFlag.SRM_ELK_LOGS_V2)
+  const isELKEnabled = useFeatureFlag(FeatureFlag.SRM_ELK_LOGS_V2) || true
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/customiseHealthSource/CustomiseHealthSource.utils.tsx:60]: are we keeping enabled by default ?

File: src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts [hunk 2]
```diff
@@ -21,3 +23,14 @@ export function getIsQueryFieldNotPresent(
 ): boolean {
   return Boolean(queryFieldIdentifier && !values[queryFieldIdentifier as keyof CommonCustomMetricFormikInterface])
 }
+
+export function getRunQueryBtnTooltip(
+  queryFieldIdentifier: string | undefined,
+  values: CommonCustomMetricFormikInterface,
+  query: string,
+  getString: UseStringsReturn['getString']
+): string {
```
Comment (deepesh-ui) [inline src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts:32]: do we need complete values just to determine tooltip ?

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/customiseHealthSource/CustomiseHealthSource.utils.tsx (hunk 1 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/customiseHealthSource/CustomiseHealthSource.utils.tsx]: corrected.

File: src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts (hunk 2 above)
Comment (harshilbits) [reply src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts:32]: yeah `values` are used in child function also in a dynamic way.

Comment (deepesh-ui) [reply src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts:32]: we are using only one property from values right ? So we can move that check here

**Review by deepesh-ui — APPROVED** [review]

**Pull request merged**